common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.h

```diff
@@ -98,8 +98,8 @@ class mon_wlan_hal_dwpal : public base_wlan_hal_dwpal, public mon_wlan_hal {
 
     bool dwpal_set_scan_params_fg(const sScanCfgParams &params)
     {
-        if (dwpal_nl_cmd_set(m_radio_info.iface_name, LTQ_NL80211_VENDOR_SUBCMD_SET_SCAN_PARAMS,
-                             (unsigned char *)&params, sizeof(params))) {
+        if (!dwpal_nl_cmd_set(m_radio_info.iface_name, LTQ_NL80211_VENDOR_SUBCMD_SET_SCAN_PARAMS,
```

Comment:
This looks like it should be a separate commit or at least explained in the commit message.